src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetVersionImportService.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Public.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.Database;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services.Public.Data;
+
+public class DataSetVersionImportService(
+    ContentDbContext contentDbContext,
+    PublicDataDbContext publicDataDbContext) 
+    : IDataSetVersionImportService
+{
+    public async Task<bool> IsPublicApiDataSetImportsInProgress(Guid releaseVersionId)
+    {
+        var dataFileIds = await contentDbContext
+            .ReleaseFiles
+            .Where(rf => rf.ReleaseVersionId == releaseVersionId && rf.File.Type == FileType.Data)
+            .Select(rf => rf.FileId)
+            .ToListAsync();
+
+        return await publicDataDbContext
+            .DataSetVersionImports
+            .AsNoTracking()
+            .AnyAsync(import => 
+                dataFileIds.Contains(import.DataSetVersion.CsvFileId) 
+                && !DataSetVersionImportStatusConstants.TerminalStates.ToList().Contains(import.Status));
```

Comment:
I think in future if we allow an import to be retried for the same dataset version this would not work as we'd have a history of imports for the same version and  only really need to consider the latest import.

I'm wondering if we really need to be concerned with the `DataSetVersionImport` though and be filtering by a list of the terminal states for the import, when `DataSetVersion` has a status where we can just check it's not still `DataSetVersionStatus.Processing` or `DataSetVersionStatus.Failed`.

Reply:
Updated to use DataSetVersion.Status.